go/pserver/service.go

```diff
@@ -137,11 +141,8 @@ func LoadCheckpoint(e *EtcdClient, idx int) (Checkpoint, error) {
 		return nil, err
 	}
 
-	// TODO(helin): change MD5 to CRC since CRC is better for file
-	// checksum in our use case (emphasize speed over security).
-	h := md5.New()
-	md5 := hex.EncodeToString(h.Sum(content))
-	if md5 != cpMeta.MD5 {
+	crc32 := crc32.ChecksumIEEE(content)
```

Comment:
Curious why md5 will cause the error?

Reply:
The same question...

Reply:
@typhoonzero @Yancey1989 @dzhwinter 

The problem was I think we used the md5 package incorrectly. To show to error:
```go
package main

import (
	"crypto/md5"
	"encoding/hex"
	"fmt"
)

func main() {
	h := md5.New()
	md5 := hex.EncodeToString(h.Sum([]byte("hello this is some string")))
        // Output: 68656c6c6f207468697320697320736f6d6520737472696e67d41d8cd98f00b204e9800998ecf8427e
	fmt.Println(md5)
}
```
The output is not a typical MD5 string, rather a very long one, and causes etcd write error with "message too large".

I think the correct way to get the MD5 string is here:
```go
package main

import (
	"crypto/md5"
	"fmt"
)

func main() {
	data := []byte("These pretzels are making me thirsty.")
	fmt.Printf("%x", md5.Sum(data))
}
```

The reason to switch to CRC32 is because it's faster, better for checksum, MD5 is slower, better for defending cracking.